tests/pytest_framework/src/syslog_ng/syslog_ng_cli.py

```diff
@@ -54,30 +54,51 @@ def __syntax_only(self, config_path=None):
             command_short_name="syntax_only", command=["--syntax-only", "--cfgfile={}".format(config_path)]
         )
 
-    # Process commands
-    def start(self, config):
-        self.__logger.info("Beginning of syslog-ng start")
-        config.write_config_content()
-
-        # syntax check
+    def __syntax_check(self):
         result = self.__syntax_only()
         if result["exit_code"] != 0:
             self.__logger.error(result["stderr"])
             raise Exception("syslog-ng can not started")
 
-        # effective start
-        if self.__valgrind_usage:
-            self.__process = self.__syslog_ng_executor.run_process_with_valgrind()
-        else:
-            self.__process = self.__syslog_ng_executor.run_process()
-
+    def __is_process_running(self):
+        return self.__process.poll() == None
+
+    def wait_for_control_socket_alive(self):
+        class StoppedUnexpectedly(Exception):
```

Comment:
I would not create separate Exception class for now. We already have several raise calls with standard Exception class, they would be strange to read.
If we wanted to add description about the raise we can use Exception("why we raises") form. I understand this is a convention. But if we started to create classes for raising let collect them in separate files, and we can iteratively extend them. I wanted to structure pytest_framework as clear as possible.
My other comment to here: 
I would not create wait_for_control_socket_alive() in this way. What we wanted here is to wait if syslog-ng process can started or not with: self.process.poll(), this is welcome.
But, I would create only a new class function:
```
def __wait_for_process_alive(self): 
    return wait_until_true(__is_process_running)
```
Maybe __is_process_running() also should be contains a check with an inside raise (as you wrote).
But, after this we can use it like (I think it is much readable):
```
    def wait_for_start(self):
        # wait for start and check start result
        if not self.wait_for_process_alive():
            self.__error_handling()
        if not self.wait_for_control_socket_alive():
            self.__error_handling()
        ...
```
As I think it should do the same as you wanted

Ok, after this I realize there are some repetative code what we can refactor into one function:
- we wanted to wait for process, for socket, for console log message.

Reply:
Unfortunately `wait_for_process_alive` would not help here, because it will instantly return, because the process is not started. We to look for "not started" instead of "started" somehow.

With @mitzkia help, the inner class is eliminated.

Resolving the issue.